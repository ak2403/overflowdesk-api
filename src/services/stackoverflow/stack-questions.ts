import { get } from "../../helpers/fetch";
import {
  QuestionResponse,
  StackOverflowResponse,
} from "../../types/services/stackoverflow/question";
import {
  QueryOptions,
  StackOverflowApiType,
} from "../../types/services/stackoverflow";
import { structApiUrl } from "./utils";
import { StackOverflow } from "./stack-overflow";

export class StackQuestions extends StackOverflow {
  private type: string = StackOverflowApiType.Questions;

  constructor(url: string) {
    super(url);
  }

  async list(
    queryOptions: QueryOptions
  ): Promise<StackOverflowResponse<QuestionResponse>> {
    const apiUrl = structApiUrl(
      { url: this._apiUrl, type: this.type },
      queryOptions
    );

    const response = await get<StackOverflowResponse<QuestionResponse>>(apiUrl);

    return response.data;
  }

  async getById(
    id: string,
    queryOptions: QueryOptions
  ): Promise<StackOverflowResponse<QuestionResponse>> {
    const apiUrl = structApiUrl(
      { url: this._apiUrl, type: this.type, id },
      queryOptions
    );

    const response = await get<StackOverflowResponse<QuestionResponse>>(apiUrl);

    return response.data;
  }
}
