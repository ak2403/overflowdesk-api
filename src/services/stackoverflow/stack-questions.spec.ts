import { StackQuestions } from "./stack-questions";

const StackApiUrl = process.env.STACK_API_URL || "";
const StackApiKey = process.env.STACK_API_KEY || "";

describe("StackQuestions", () => {
  let stackQuestions = new StackQuestions(StackApiUrl);

  describe("list()", () => {
    it("returns response for a valid call", async () => {
      const result = await stackQuestions.list({
        key: StackApiKey,
        site: "stackoverflow",
      });

      expect(Object.keys(result)).toEqual([
        "items",
        "has_more",
        "quota_max",
        "quota_remaining",
      ]);
      expect(result).toMatchSnapshot();
    });
  });

  describe("getById()", () => {
    it("returns response for a valid id", async () => {
      const result = await stackQuestions.getById("123456", {
        key: StackApiKey,
        site: "stackoverflow",
      });

      expect(Object.keys(result)).toEqual([
        "items",
        "has_more",
        "quota_max",
        "quota_remaining",
      ]);
      expect(result).toMatchSnapshot();
    });
  });
});
