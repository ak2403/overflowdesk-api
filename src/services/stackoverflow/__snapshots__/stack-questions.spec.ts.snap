// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackQuestions getById() returns response for a valid id 1`] = `
{
  "has_more": false,
  "items": [
    {
      "answer_count": 4,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1547025378,
      "is_answered": true,
      "last_activity_date": 1724405249,
      "last_edit_date": 1547040537,
      "link": "https://stackoverflow.com/questions/54106638/userwarning-cannot-parse-header-or-footer-so-it-will-be-ignored-on-loading-xl",
      "owner": {
        "account_id": 15057646,
        "display_name": "Sam",
        "link": "https://stackoverflow.com/users/10867456/sam",
        "profile_image": "https://lh4.googleusercontent.com/-EQhkT7f9Yjo/AAAAAAAAAAI/AAAAAAAAAAA/AKxrwcanpvAGTscfl3p2inYXvksZIqcyYw/mo/photo.jpg?sz=256",
        "reputation": 61,
        "user_id": 10867456,
        "user_type": "registered",
      },
      "question_id": 54106638,
      "score": 5,
      "tags": [
        "python",
        "python-2.7",
        "openpyxl",
        "xlsm",
      ],
      "title": "UserWarning: &quot;Cannot parse header or footer so it will be ignored&quot; on loading xlsm file",
      "view_count": 19079,
    },
  ],
  "quota_max": 10000,
  "quota_remaining": 9957,
}
`;

exports[`StackQuestions list() returns response for a valid call 1`] = `
{
  "has_more": true,
  "items": [
    {
      "answer_count": 2,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724374243,
      "is_answered": true,
      "last_activity_date": 1724397488,
      "last_edit_date": 1724374434,
      "link": "https://stackoverflow.com/questions/78903981/regex-matches-specific-url-patterns-in-c",
      "owner": {
        "account_id": 7671172,
        "display_name": "carlosbou",
        "link": "https://stackoverflow.com/users/5813364/carlosbou",
        "profile_image": "https://www.gravatar.com/avatar/ed560c70d9daaf9cf28691bafbfac6cf?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 3,
        "user_id": 5813364,
        "user_type": "registered",
      },
      "question_id": 78903981,
      "score": 0,
      "tags": [
        "c#",
        "regex",
      ],
      "title": "Regex matches specific url patterns in C#",
      "view_count": 43,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397483,
      "is_answered": false,
      "last_activity_date": 1724397483,
      "link": "https://stackoverflow.com/questions/78904736/in-sst-v3-ion-what-is-the-equivalent-to-a-customresource",
      "owner": {
        "accept_rate": 75,
        "account_id": 136312,
        "display_name": "nillls",
        "link": "https://stackoverflow.com/users/339724/nillls",
        "profile_image": "https://www.gravatar.com/avatar/5059c4409c3889f8bce07fa62f424880?s=256&d=identicon&r=PG",
        "reputation": 623,
        "user_id": 339724,
        "user_type": "registered",
      },
      "question_id": 78904736,
      "score": 0,
      "tags": [
        "serverless",
        "pulumi",
      ],
      "title": "In SST v3 (Ion) what is the equivalent to a CustomResource?",
      "view_count": 2,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397457,
      "is_answered": false,
      "last_activity_date": 1724397457,
      "link": "https://stackoverflow.com/questions/78904735/how-to-extend-eslint-config-when-using-yarn-workspaces",
      "owner": {
        "account_id": 18758881,
        "display_name": "howard wolowitz",
        "link": "https://stackoverflow.com/users/13678259/howard-wolowitz",
        "profile_image": "https://lh6.googleusercontent.com/-C0VbgVw5Y_E/AAAAAAAAAAI/AAAAAAAAAAA/AMZuuckR0gaL5t70zIPk_f2hfcn-W_zi_g/photo.jpg?sz=256",
        "reputation": 611,
        "user_id": 13678259,
        "user_type": "registered",
      },
      "question_id": 78904735,
      "score": 0,
      "tags": [
        "javascript",
        "reactjs",
        "eslint",
        "hadoop-yarn",
      ],
      "title": "How to extend eslint config when using yarn workspaces",
      "view_count": 4,
    },
    {
      "accepted_answer_id": 72188902,
      "answer_count": 5,
      "content_license": "CC BY-SA 3.0",
      "creation_date": 1496741225,
      "is_answered": true,
      "last_activity_date": 1724397456,
      "last_edit_date": 1496749721,
      "link": "https://stackoverflow.com/questions/44386205/how-to-monitor-the-size-of-a-directory-via-telegraf",
      "owner": {
        "accept_rate": 100,
        "account_id": 3277907,
        "display_name": "Tw Bert",
        "link": "https://stackoverflow.com/users/2759336/tw-bert",
        "profile_image": "https://i.sstatic.net/MU3ba.png?s=256",
        "reputation": 3759,
        "user_id": 2759336,
        "user_type": "registered",
      },
      "question_id": 44386205,
      "score": 9,
      "tags": [
        "du",
        "telegraf",
      ],
      "title": "How to monitor the size of a directory via Telegraf",
      "view_count": 11430,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724394733,
      "is_answered": false,
      "last_activity_date": 1724397451,
      "last_edit_date": 1724397451,
      "link": "https://stackoverflow.com/questions/78904578/valid-types-for-terminal-screen-in-whatsapp-flows",
      "owner": {
        "account_id": 29138056,
        "display_name": "Pieter le Roux",
        "link": "https://stackoverflow.com/users/22321741/pieter-le-roux",
        "profile_image": "https://lh3.googleusercontent.com/a/AAcHTtfJDZ33oU5IddNayA12ThL9tk-dTT2z0BwIxDt2gW3_cYs=k-s256",
        "reputation": 45,
        "user_id": 22321741,
        "user_type": "registered",
      },
      "question_id": 78904578,
      "score": 0,
      "tags": [
        "whatsapp-flows",
      ],
      "title": "Valid types for Terminal screen in WhatsApp Flows",
      "view_count": 7,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1317916812,
      "is_answered": false,
      "last_activity_date": 1724397448,
      "last_edit_date": 1724397448,
      "link": "https://stackoverflow.com/questions/7677101/how-can-i-set-the-paginator-page-from-within-the-controller-in-cakephp",
      "owner": {
        "account_id": 64027,
        "display_name": "GTM",
        "link": "https://stackoverflow.com/users/189218/gtm",
        "profile_image": "https://www.gravatar.com/avatar/6bab40b6cd391e776a4631888101766f?s=256&d=identicon&r=PG",
        "reputation": 61,
        "user_id": 189218,
        "user_type": "registered",
      },
      "question_id": 7677101,
      "score": 3,
      "tags": [
        "cakephp",
        "pagination",
        "paginator",
      ],
      "title": "How can I set the paginator page from within the controller in cakePHP",
      "view_count": 2940,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724394917,
      "is_answered": false,
      "last_activity_date": 1724397443,
      "last_edit_date": 1724395670,
      "link": "https://stackoverflow.com/questions/78904590/asp-net-mvc-application-failed-to-start",
      "owner": {
        "accept_rate": 61,
        "account_id": 1461614,
        "display_name": "tahasozgen",
        "link": "https://stackoverflow.com/users/1375780/tahasozgen",
        "profile_image": "https://lh5.googleusercontent.com/-ZaySVPzbyv0/AAAAAAAAAAI/AAAAAAAAAEE/U6JfJjWJ1gk/photo.jpg?sz=256",
        "reputation": 491,
        "user_id": 1375780,
        "user_type": "registered",
      },
      "question_id": 78904590,
      "score": 0,
      "tags": [
        "c#",
        "asp.net-mvc-4",
      ],
      "title": "Asp.Net MVC Application Failed to Start",
      "view_count": 20,
    },
    {
      "answer_count": 2,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724271198,
      "is_answered": false,
      "last_activity_date": 1724397430,
      "link": "https://stackoverflow.com/questions/78898845/merge-touching-polygons-within-sf-dataframe-or-terra-spatvector-in-r",
      "owner": {
        "account_id": 15227255,
        "display_name": "Jake L",
        "link": "https://stackoverflow.com/users/10987450/jake-l",
        "profile_image": "https://graph.facebook.com/10216303729499548/picture?type=large",
        "reputation": 1047,
        "user_id": 10987450,
        "user_type": "registered",
      },
      "question_id": 78898845,
      "score": 1,
      "tags": [
        "r",
        "merge",
        "spatial",
        "terra",
      ],
      "title": "Merge touching polygons within sf dataframe or terra spatvector in R",
      "view_count": 35,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724396275,
      "is_answered": false,
      "last_activity_date": 1724397407,
      "link": "https://stackoverflow.com/questions/78904658/navigation-bar-swiftui-toolbar-background",
      "owner": {
        "account_id": 20710474,
        "display_name": "Mykyta Shapko",
        "link": "https://stackoverflow.com/users/15207216/mykyta-shapko",
        "profile_image": "https://lh3.googleusercontent.com/-hf9Sm0j8-CE/AAAAAAAAAAI/AAAAAAAAAAA/AMZuucn-SDe0bRhbUqzabGA-pp4U3ZAa6w/s96-c/photo.jpg?sz=256",
        "reputation": 33,
        "user_id": 15207216,
        "user_type": "registered",
      },
      "question_id": 78904658,
      "score": 0,
      "tags": [
        "swift",
        "swiftui",
        "navigationbar",
      ],
      "title": "Navigation bar SwiftUI toolbar background",
      "view_count": 13,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724030389,
      "is_answered": true,
      "last_activity_date": 1724397404,
      "link": "https://stackoverflow.com/questions/78886026/pandas-kurtosis-skew-bug",
      "owner": {
        "account_id": 12326812,
        "display_name": "z xt",
        "link": "https://stackoverflow.com/users/8994215/z-xt",
        "profile_image": "https://www.gravatar.com/avatar/a24319ad63d295bbdb095e651b4c9cff?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 49,
        "user_id": 8994215,
        "user_type": "registered",
      },
      "question_id": 78886026,
      "score": 0,
      "tags": [
        "python",
        "pandas",
        "numpy",
      ],
      "title": "pandas kurtosis/skew bug?",
      "view_count": 70,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397394,
      "is_answered": false,
      "last_activity_date": 1724397394,
      "link": "https://stackoverflow.com/questions/78904732/angularjs-do-not-get-data-from-symfony-sometimes",
      "owner": {
        "account_id": 6606515,
        "display_name": "Lenw&#235; Galathil",
        "link": "https://stackoverflow.com/users/5103184/lenw%c3%ab-galathil",
        "profile_image": "https://graph.facebook.com/10207223080365461/picture?type=large",
        "reputation": 49,
        "user_id": 5103184,
        "user_type": "registered",
      },
      "question_id": 78904732,
      "score": 0,
      "tags": [
        "angularjs",
        "api",
        "http",
      ],
      "title": "AngularJS do not get data from Symfony sometimes",
      "view_count": 3,
    },
    {
      "accepted_answer_id": 2239718,
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1265026301,
      "is_answered": true,
      "last_activity_date": 1724397383,
      "last_edit_date": 1724397383,
      "link": "https://stackoverflow.com/questions/2176479/datapager-and-datagrid-how-to-move-to-the-page-of-current-selecteditem",
      "owner": {
        "accept_rate": 67,
        "account_id": 57609,
        "display_name": "Lars Udengaard",
        "link": "https://stackoverflow.com/users/172603/lars-udengaard",
        "profile_image": "https://www.gravatar.com/avatar/237cb805f3e28755f58e0ddd3cfba13b?s=256&d=identicon&r=PG",
        "reputation": 1257,
        "user_id": 172603,
        "user_type": "registered",
      },
      "question_id": 2176479,
      "score": 0,
      "tags": [
        ".net",
        "pagination",
        "datagrid",
        "datapager",
      ],
      "title": "DataPager and Datagrid, How to move to the page of current selecteditem",
      "view_count": 1774,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397353,
      "is_answered": false,
      "last_activity_date": 1724397353,
      "link": "https://stackoverflow.com/questions/78904729/how-to-obtain-an-adfs-token-for-microsoft-graph-api-in-postman",
      "owner": {
        "account_id": 31991529,
        "display_name": "손동진",
        "link": "https://stackoverflow.com/users/24799965/%ec%86%90%eb%8f%99%ec%a7%84",
        "profile_image": "https://www.gravatar.com/avatar/2cc2fab1a9201e06d4b2dc9e33ef6d00?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 33,
        "user_id": 24799965,
        "user_type": "registered",
      },
      "question_id": 78904729,
      "score": 0,
      "tags": [
        "authentication",
        "oauth-2.0",
        "microsoft-graph-api",
        "postman",
        "adfs",
      ],
      "title": "How to Obtain an ADFS Token for Microsoft Graph API in Postman?",
      "view_count": 9,
    },
    {
      "answer_count": 4,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1723788692,
      "is_answered": true,
      "last_activity_date": 1724397341,
      "link": "https://stackoverflow.com/questions/78877805/calculating-an-average-of-multiple-polygons",
      "owner": {
        "accept_rate": 73,
        "account_id": 76593,
        "display_name": "Paul Jurczak",
        "link": "https://stackoverflow.com/users/219153/paul-jurczak",
        "profile_image": "https://i.sstatic.net/je9qW.jpg?s=256",
        "reputation": 7858,
        "user_id": 219153,
        "user_type": "registered",
      },
      "question_id": 78877805,
      "score": 1,
      "tags": [
        "python",
        "computational-geometry",
        "shapely",
      ],
      "title": "Calculating an average of multiple polygons",
      "view_count": 110,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724344581,
      "is_answered": false,
      "last_activity_date": 1724397340,
      "last_edit_date": 1724389102,
      "link": "https://stackoverflow.com/questions/78902642/azure-blob-trigger-function-not-triggering-once-deployed",
      "owner": {
        "accept_rate": 29,
        "account_id": 2982061,
        "display_name": "Muhammad Ashhar Hasan",
        "link": "https://stackoverflow.com/users/2531826/muhammad-ashhar-hasan",
        "profile_image": "https://i.sstatic.net/Xf8hQ.jpg?s=256",
        "reputation": 299,
        "user_id": 2531826,
        "user_type": "registered",
      },
      "question_id": 78902642,
      "score": 0,
      "tags": [
        "azure",
        "azure-functions",
        "azure-blob-storage",
        "azure-blob-trigger",
      ],
      "title": "Azure Blob Trigger Function not triggering once deployed",
      "view_count": 25,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724311863,
      "is_answered": false,
      "last_activity_date": 1724397317,
      "last_edit_date": 1724321590,
      "link": "https://stackoverflow.com/questions/78900270/this-is-a-question-about-tkinters-configure-binding",
      "owner": {
        "account_id": 27568959,
        "display_name": "kimhyunju",
        "link": "https://stackoverflow.com/users/21040543/kimhyunju",
        "profile_image": "https://lh3.googleusercontent.com/a/AEdFTp5jDxz0RlNi6F2HNhhprWVvbq5W1BnKuzxm50LQ=k-s256",
        "reputation": 334,
        "user_id": 21040543,
        "user_type": "registered",
      },
      "question_id": 78900270,
      "score": 0,
      "tags": [
        "python",
        "tkinter",
      ],
      "title": "This is a question about tkinter&#39;s &lt;Configure&gt; binding",
      "view_count": 36,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724394734,
      "is_answered": false,
      "last_activity_date": 1724397308,
      "last_edit_date": 1724396713,
      "link": "https://stackoverflow.com/questions/78904579/how-do-i-rotate-a-uv-texture-on-a-sphere-without-tilting-it",
      "owner": {
        "account_id": 23388684,
        "display_name": "thabnir",
        "link": "https://stackoverflow.com/users/17453615/thabnir",
        "profile_image": "https://i.sstatic.net/zs76Q.jpg?s=256",
        "reputation": 11,
        "user_id": 17453615,
        "user_type": "registered",
      },
      "question_id": 78904579,
      "score": 0,
      "tags": [
        "rust",
        "graphics",
        "textures",
        "linear-algebra",
        "nalgebra",
      ],
      "title": "How do I rotate a uv texture on a sphere without tilting it?",
      "view_count": 13,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397281,
      "is_answered": false,
      "last_activity_date": 1724397281,
      "link": "https://stackoverflow.com/questions/78904724/exception-in-monitor-thread-while-connecting-to-server-localhost27017-com-mongo",
      "owner": {
        "account_id": 9782480,
        "display_name": "Nani",
        "link": "https://stackoverflow.com/users/7249929/nani",
        "profile_image": "https://www.gravatar.com/avatar/7aea9636ca29850fa5cc0bffd0227aeb?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 65,
        "user_id": 7249929,
        "user_type": "registered",
      },
      "question_id": 78904724,
      "score": 0,
      "tags": [
        "java",
        "spring",
        "mongodb",
        "spring-boot",
      ],
      "title": "Exception in monitor thread while connecting to server localhost:27017 com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message",
      "view_count": 7,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724383480,
      "is_answered": false,
      "last_activity_date": 1724397279,
      "last_edit_date": 1724397279,
      "link": "https://stackoverflow.com/questions/78904203/failed-in-using-buttons-as-tabs-in-shiny",
      "owner": {
        "account_id": 27464941,
        "display_name": "Snow ",
        "link": "https://stackoverflow.com/users/20954835/snow",
        "profile_image": "https://lh3.googleusercontent.com/a/AEdFTp5vwkB7p4C-yK7MPU8F2KPadjqqpp4H7PUc9fdg=k-s256",
        "reputation": 11,
        "user_id": 20954835,
        "user_type": "registered",
      },
      "question_id": 78904203,
      "score": 0,
      "tags": [
        "plotly",
        "shinyapps",
        "renderui",
      ],
      "title": "Failed in Using Buttons as Tabs in Shiny",
      "view_count": 18,
    },
    {
      "answer_count": 19,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1613418755,
      "is_answered": true,
      "last_activity_date": 1724397265,
      "link": "https://stackoverflow.com/questions/66214555/gradle-sync-failed-removecontententry-removed-content-entry-url-still-ex",
      "owner": {
        "account_id": 6747565,
        "display_name": "Stefan",
        "link": "https://stackoverflow.com/users/5198699/stefan",
        "profile_image": "https://www.gravatar.com/avatar/00e5f318f04c4adc79a5d8a6c1592243?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 531,
        "user_id": 5198699,
        "user_type": "registered",
      },
      "question_id": 66214555,
      "score": 53,
      "tags": [
        "android",
        "gradle",
        "intellij-idea",
      ],
      "title": "Gradle sync failed: removeContentEntry: removed content entry url &#39;...&#39; still exists after removing",
      "view_count": 20918,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724385173,
      "is_answered": false,
      "last_activity_date": 1724397264,
      "link": "https://stackoverflow.com/questions/78904249/how-do-i-make-multiple-unity-buttons-with-unityactions-that-depend-on-a-variable",
      "owner": {
        "account_id": 14087595,
        "display_name": "Robin",
        "link": "https://stackoverflow.com/users/10509565/robin",
        "profile_image": "https://www.gravatar.com/avatar/03c54518a0e0e07e0585703e80cc69d9?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 385,
        "user_id": 10509565,
        "user_type": "registered",
      },
      "question_id": 78904249,
      "score": 0,
      "tags": [
        "c#",
        "unity-game-engine",
        "user-interface",
      ],
      "title": "How do I make multiple Unity buttons with UnityActions that depend on a variable?",
      "view_count": 15,
    },
    {
      "answer_count": 1,
      "bounty_amount": 50,
      "bounty_closes_date": 1724401672,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1723113027,
      "is_answered": false,
      "last_activity_date": 1724397254,
      "last_edit_date": 1723796958,
      "link": "https://stackoverflow.com/questions/78847901/spring-boot-rabbitmq-rotate-credentials",
      "owner": {
        "accept_rate": 25,
        "account_id": 1786097,
        "display_name": "P&#233;&#233;",
        "link": "https://stackoverflow.com/users/1627542/p%c3%a9%c3%a9",
        "profile_image": "https://www.gravatar.com/avatar/71a1be29d4534c0a063258ae7cdaf011?s=256&d=identicon&r=PG",
        "reputation": 53,
        "user_id": 1627542,
        "user_type": "registered",
      },
      "question_id": 78847901,
      "score": 2,
      "tags": [
        "spring",
        "spring-boot",
        "rabbitmq",
        "credentials",
        "amqp",
      ],
      "title": "Spring-boot RabbitMQ rotate credentials",
      "view_count": 84,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397250,
      "is_answered": false,
      "last_activity_date": 1724397250,
      "link": "https://stackoverflow.com/questions/78904720/handle-background-task-in-watchos",
      "owner": {
        "account_id": 7233076,
        "display_name": "nithindev.n",
        "link": "https://stackoverflow.com/users/5519224/nithindev-n",
        "profile_image": "https://www.gravatar.com/avatar/7c4aa56f78ca73ee1ff491e6a9db2f44?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 49,
        "user_id": 5519224,
        "user_type": "registered",
      },
      "question_id": 78904720,
      "score": 0,
      "tags": [
        "ios",
        "swift",
        "swiftui",
        "watchos",
        "background-task",
      ],
      "title": "Handle Background task in watchOS",
      "view_count": 6,
    },
    {
      "answer_count": 3,
      "content_license": "CC BY-SA 3.0",
      "creation_date": 1490019686,
      "is_answered": false,
      "last_activity_date": 1724397235,
      "last_edit_date": 1490222860,
      "link": "https://stackoverflow.com/questions/42906055/angular-ng-class-working-on-td-but-not-on-tr",
      "owner": {
        "account_id": 10501873,
        "display_name": "sborn",
        "link": "https://stackoverflow.com/users/7740356/sborn",
        "profile_image": "https://www.gravatar.com/avatar/12cb121d62239b563ab81d1b1158ae10?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 121,
        "user_id": 7740356,
        "user_type": "registered",
      },
      "question_id": 42906055,
      "score": 0,
      "tags": [
        "html",
        "css",
        "angularjs",
        "html-table",
      ],
      "title": "Angular - ng-class working on td but not on tr",
      "view_count": 2381,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397208,
      "is_answered": false,
      "last_activity_date": 1724397208,
      "link": "https://stackoverflow.com/questions/78904719/azureaks-monitoring-and-alerting-check-whether-pods-for-the-application-that-a",
      "owner": {
        "account_id": 11976033,
        "display_name": "Ranjeet",
        "link": "https://stackoverflow.com/users/8763127/ranjeet",
        "profile_image": "https://i.sstatic.net/Whxw5.png?s=256",
        "reputation": 49,
        "user_id": 8763127,
        "user_type": "registered",
      },
      "question_id": 78904719,
      "score": 0,
      "tags": [
        "azure",
        "kubernetes",
        "azure-functions",
        "azure-aks",
      ],
      "title": "Azure(AKS) Monitoring and Alerting check whether pods for the application that are active and listening to agents. Please guide me",
      "view_count": 6,
    },
    {
      "answer_count": 1,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724334699,
      "is_answered": false,
      "last_activity_date": 1724397191,
      "link": "https://stackoverflow.com/questions/78901932/app-mapdynamiccontrollerrouterouterpath-doesnt-seem-to-affect-routing",
      "owner": {
        "accept_rate": 80,
        "account_id": 287571,
        "display_name": "Tar",
        "link": "https://stackoverflow.com/users/587467/tar",
        "profile_image": "https://www.gravatar.com/avatar/7787bdb4718fdc9b42fb46d671bd7eca?s=256&d=identicon&r=PG",
        "reputation": 8913,
        "user_id": 587467,
        "user_type": "registered",
      },
      "question_id": 78901932,
      "score": 0,
      "tags": [
        "c#",
        "asp.net",
        "routes",
        ".net-8.0",
        "minimal-apis",
      ],
      "title": "app.MapDynamicControllerRoute&lt;Router&gt;(&quot;{**path}&quot;) doesn&#39;t seem to affect routing",
      "view_count": 16,
    },
    {
      "accepted_answer_id": 54757246,
      "answer_count": 4,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1550535588,
      "is_answered": true,
      "last_activity_date": 1724397178,
      "link": "https://stackoverflow.com/questions/54757225/how-to-switch-from-terminal-back-to-editor-in-vim",
      "owner": {
        "accept_rate": 90,
        "account_id": 9751083,
        "display_name": "mle0312",
        "link": "https://stackoverflow.com/users/7899820/mle0312",
        "profile_image": "https://www.gravatar.com/avatar/fa57d673eb787ff6554f75313a1a7221?s=256&d=identicon&r=PG&f=y&so-version=2",
        "reputation": 415,
        "user_id": 7899820,
        "user_type": "registered",
      },
      "question_id": 54757225,
      "score": 6,
      "tags": [
        "linux",
        "vim",
      ],
      "title": "How to switch from terminal back to editor in vim?",
      "view_count": 13267,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397175,
      "is_answered": false,
      "last_activity_date": 1724397175,
      "link": "https://stackoverflow.com/questions/78904716/how-can-i-make-data-values-offset-from-markers",
      "owner": {
        "account_id": 20065588,
        "display_name": "Beket Kassymbekov",
        "link": "https://stackoverflow.com/users/14711329/beket-kassymbekov",
        "profile_image": "https://lh3.googleusercontent.com/a-/AOh14Gj5mLMkvxGo_f9BzEZdVcj5UQgdcUFgIHKeAWqP2K4=k-s256",
        "reputation": 33,
        "user_id": 14711329,
        "user_type": "registered",
      },
      "question_id": 78904716,
      "score": 0,
      "tags": [
        "python",
      ],
      "title": "How can I make Data values offset from markers?",
      "view_count": 7,
    },
    {
      "answer_count": 0,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1724397152,
      "is_answered": false,
      "last_activity_date": 1724397152,
      "link": "https://stackoverflow.com/questions/78904715/yolov10-how-to-interpret-training-progress-info",
      "owner": {
        "accept_rate": 93,
        "account_id": 997598,
        "display_name": "Putnik",
        "link": "https://stackoverflow.com/users/1013183/putnik",
        "profile_image": "https://www.gravatar.com/avatar/458baa81702bf54b047957e27daa5673?s=256&d=identicon&r=PG",
        "reputation": 6536,
        "user_id": 1013183,
        "user_type": "registered",
      },
      "question_id": 78904715,
      "score": 0,
      "tags": [
        "computer-vision",
        "yolo",
      ],
      "title": "YOLOv10: how to interpret training progress info?",
      "view_count": 4,
    },
    {
      "accepted_answer_id": 72151584,
      "answer_count": 5,
      "content_license": "CC BY-SA 4.0",
      "creation_date": 1651918587,
      "is_answered": true,
      "last_activity_date": 1724397151,
      "link": "https://stackoverflow.com/questions/72151550/unable-to-install-redis-extension-in-docker-php-fpm-alpine-image",
      "owner": {
        "accept_rate": 33,
        "account_id": 5809273,
        "display_name": "Mimmo",
        "link": "https://stackoverflow.com/users/4580920/mimmo",
        "profile_image": "https://i.sstatic.net/P3d3s.jpg?s=256",
        "reputation": 123,
        "user_id": 4580920,
        "user_type": "registered",
      },
      "question_id": 72151550,
      "score": 0,
      "tags": [
        "docker",
        "redis",
        "alpine-linux",
      ],
      "title": "Unable to install Redis extension in Docker php fpm-alpine image",
      "view_count": 11161,
    },
  ],
  "quota_max": 10000,
  "quota_remaining": 9991,
}
`;
